ui/app/routes/workspace/projects/project/app/build.ts

```diff
@@ -28,11 +28,12 @@ export default class BuildDetail extends Route {
 
   async model(params: BuildModelParams) {
     // Setup the build request
+    let { builds } = this.modelFor('workspace.projects.project.app');
+    let { id: build_id } = builds.find((obj) => obj.sequence == params.sequence);
     let ref = new Ref.Operation();
-    ref.setId(params.build_id);
+    ref.setId(build_id);
     let req = new GetBuildRequest();
     req.setRef(ref);
-
     let build = await this.api.client.getBuild(req, this.api.WithMeta());
     return build.toObject();
```

Comment:
This approach is totally great and very pragmatic 👍 

[dust] For completeness (and apologies if you’re already aware of this) it’s also possible to use `GetBuild` with a sequence number. It’d look something like this:

```ts
async model(params: BuildModelParams): Promise<Build.AsObject> {
  let { sequence } = params;
  let { project_id: projectName } = this.paramsFor('workspace.projects.project') as ProjectRouteParams;
  let { app_id: appName } = this.paramsFor('workspace.projects.project.app') as AppRouteParams;

  let req = new GetBuildRequest();
  let ref = new Ref.Operation();
  let appRef = new Ref.Application();
  let seqRef = new Ref.OperationSeq();

  appRef.setProject(projectName);
  appRef.setApplication(appName);

  seqRef.setApplication(appRef);
  seqRef.setNumber(sequence);

  ref.setSequence(seqRef);

  req.setRef(ref);

  let build = await this.api.client.getBuild(req, this.api.WithMeta());

  return build.toObject();
}
```

This is true of `GetDeployment` and `GetRelease` too. Let’s not use this now, but it’s useful to know it’s available should we need it in future.

Reply:
Didn't realize this, or maybe I found that out before I could grab values from parent components and then forgot when implementing here. Good to know in the future though!